generate-images.sh

```diff
@@ -1,32 +1,100 @@
 #!/bin/bash
-# prepare options list by searching through group_vars/ and locations/
-unset locations i
-for location in $({
-  while IFS= read -r -d $'\0' file; do
-    prefix="./locations/"
-    l=${file/#$prefix}
-    echo "${l%.yml}"
-  done < <(find ./locations/ -name "*.yml" -print0)
-  while IFS= read -r -d $'\0' directory; do
-    prefix="./group_vars/location_"
-    echo "${directory/#$prefix}"
-  done < <(find ./group_vars/ -maxdepth 1 -type d -name "location_*" -print0)
-} | sort); do
-  locations[i++]="$location"
-done
+# shellcheck disable=SC2053
+
+# Prepare options list by searching through locations/
+unset locations hosts
+
+# Extract valid locations
+mapfile -t locations < <(yq '.location' locations/*.yml | sed 's/\"//g' | sort)
+
+# Extract valid hosts
+mapfile -t hosts < <(yq '.hosts[] | select(.role == "corerouter" or .role == "gateway" or .role == "ap") | .hostname' locations/*.yml | sed 's/\"//g' | sort)
```

Comment:
You can use `yq -r` instead of the sed call: "-q  output raw strings, not JSON texts"